includes/functions.php

```diff
@@ -107,6 +107,22 @@ function getHostOS($device)
         }
     }
 
+    $pattern = $config['install_dir'] . '/includes/definitions/*.yaml';
+    foreach (glob($pattern) as $file) {
+        $tmp = Symfony\Component\Yaml\Yaml::parse(
+            file_get_contents($file)
+        );
+        if ($tmp['discovery']['method'] === 'sysObjectId' && is_array($tmp['discovery']['sysObjectId'])) {
```

Comment:
The idea with method would be to expand it out like 'sysDescr_regex' and similar (totally made up).